Assets/_Application/Layers/LayerTypes/Credentials/Properties/CredentialsInputPropertySection.cs

```diff
@@ -17,35 +19,54 @@ public class CredentialsInputPropertySection : MonoBehaviour, ILayerCredentialIn
         [SerializeField] private TMP_InputField keyTokenOrCodeInputField;
         [SerializeField] private TMP_Dropdown credentialTypeDropdown;
         private bool skipFirstCredentialErrorMessage = true;
+        private TMP_InputField inputFieldToUseForPasswordOrKey;
+
+        [SerializeField] private bool visibleOnAwake; //todo: find a better way to do this
 
-        public LayerCredentialsHandler Handler
+        public ICredentialHandler Handler
         {
             get => handler;
             set
             {
-                if (handler)
-                    handler.CredentialsAccepted.RemoveListener(OnCredentialsAccepted);
+                handler?.OnAuthorizationHandled.RemoveListener(OnCredentialsAccepted);
 
                 handler = value;
-
                 skipFirstCredentialErrorMessage = true;
 
-                handler.CredentialsAccepted.AddListener(OnCredentialsAccepted);
+                handler.OnAuthorizationHandled.AddListener(OnCredentialsAccepted);
```

Comment:
Kan de handler een null value hebben en moet hier dan een null check/null operator zitten?

Reply:
Dat zou theoretisch inderdaad kunnen, maar dat wil je nooit omdat dit de UI is die credentials ergens op moet zetten. Ik denk dat een null check hier daarom onwenselijk is, omdat het enige dat het doet de error verbergen is wat debuggen dan moeilijker maakt.